### PR TITLE
MangaDex: Clean manga descriptions

### LINKED ISSUE #10610: MangaDex: Remove Markdown tags from the description
### Describe your suggested feature

Some titles have markdown texts in their description/about and it looks realy bad on tachiyomi. 

[Orginal post](https://github.com/tachiyomiorg/tachiyomi/issues/6213)

### Other details

This can be fixed on the extension side by using some library like JetBrains/markdown to remove the Markdown tags. (Alessandro)

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I have updated the app to version **[0.12.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by curche**
Closes #10610

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by alessandrojean** [comment]
> Extra Notes usually just has links to external sites which include anime info/licensors/alternative raws; everything on other websites.

That's why I suggested that the current approach of removing them is good.

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt [hunk 1]
```diff
@@ -75,7 +75,12 @@ class MangaDexHelper() {
         while (bbRegex.containsMatchIn(intermediate)) {
             intermediate = intermediate.replace(bbRegex, "$2")
         }
-        return Parser.unescapeEntities(intermediate, false)
+        val unescapedString = Parser.unescapeEntities(intermediate, false)
+        val linkRegex = "\\[([^]]+)\\]\\(([^)]+)\\)".toRegex()
```
Comment (alessandrojean) [inline src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt:79]: Maybe make it a constant?

**Comment by CarlosEsco** [comment]
Does Tachi allow selecting text in the summary?  If so I dont see why the urls have to be removed

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt [hunk 2]
```diff
@@ -75,7 +75,12 @@ class MangaDexHelper() {
         while (bbRegex.containsMatchIn(intermediate)) {
```
Comment (CarlosEsco) [inline src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt:75]: i would strip out anything relating to bbcode/bbregex as 99% of manga have been converted to markdown

**Comment by alessandrojean** [comment]
> Does Tachi allow selecting text in the summary? If so I dont see why the urls have to be removed

It doesn't, but there's an open issue in the main repo requesting this feature, since J2K have.

**Comment by CarlosEsco** [comment]
> > Does Tachi allow selecting text in the summary? If so I dont see why the urls have to be removed
> 
> It doesn't, but there's an open issue in the main repo requesting this feature, since J2K have.

Curche can you post a screen with keeping links vs one with out?  at the end of the day they can also click the webview if they really wanted to see links

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt (hunk 2 above)
Comment (curche) [reply src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt:75]: I had my doubts but kept it for those last edge cases, if any. Will remove

File: src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt (hunk 1 above)
Comment (curche) [reply src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt:79]: Sure, makes sense to move it to a companion object.

**Comment by curche** [comment]
(Apologies to everyone for whom this doesn't render correctly)

### Opening a random manga from Latest looks fine
| Before | Now |
|--------------|-------------|
| Screenshot_20220204-002023_Tachiyomi_swan_2 | Screenshot_20220204-001908_Tachiyomi_swan_1 |

### Sometimes the old bbcode cleaning messes up and helps us out
| Before | Now |
|--------------|-------------|
| Screenshot_20220204-001408_TachiyomiSY_Mieruko_2 | Screenshot_20220204-001108_Tachiyomi_mieruko_1 |

### But what if the author had previous works that are more famous?
Readability should win over knowing it exists on (v3) MangaDex

| Before | Now |
|--------------|-------------|
| Screenshot_20220204-001430_TachiyomiSY_dedede_2 | Screenshot_20220204-001124_Tachiyomi_dedede_1 |


### Or the Manga got removed from Dex
This is very rare and its fine ~~most people wont read it and will still complain~~

| Before | Now |
|--------------|-------------|
| Screenshot_20220204-001333_TachiyomiSY_OPM_2 | Screenshot_20220204-001033_Tachiyomi_Onepiece-1 |

### Or just became an worldwide phenomenon
Ok this was too much and now it looks way better
| Before | Now |
|--------------|-------------|
| Screenshot_20220204-001255_Tachiyomi_SL_2 | Screenshot_20220204-000950_Tachiyomi_SL-1 |

**Comment by alessandrojean** [comment]
Seems that there's some empty new lines in the end, so maybe need to clean them too?

**Comment by curche** [comment]
Yeah, I'm figuring that one out. Looks like the whitespaces before notes section

**Comment by curche** [comment]
Trimmed out the trailing whitespace/newlines
Removed old bbcode cleaning code
Moved regex to companion object for better performance

PR is ready
Note that however, I haven't updated build.gradle since I'll be doing a minor PR that I prefer keeping separate but not bump extVersionCode

**Pull request merged**